### ISSUE TITLE
tcp ssl hung on connect using 2.7.9

### ISSUE THREAD

**Issue by SuiperLolo** [gevent/gevent#584]
Hello,

using gevent 1.0.2 :

- on python 2.7.6, this code is working.
- on python 2.7.9, this is not working, got a ssl handshake that never exit at client side + on_connect never called at server side.
=> python2.7/site-packages/gevent/_sslgte279.py", line 511, in do_handshake
    return self._sslobj.do_handshake()

note : you need server.crt & server.key files.

Thanks for feedback.

********************
*** 2.7.6 code
********************
```

from __future__ import print_function
from gevent import pywsgi
import gevent

from gevent import monkey
from gevent._ssl2 import SSLSocket

monkey.patch_all()


def on_connect(env, start_response):
    """
    Doc
    :param env:
    :param start_response:
    :return:
    """
    print("Connected")


def serve():
    """
    Test
    :return:
    """
    server = pywsgi.WSGIServer(('0.0.0.0', 8443), on_connect,
                               keyfile='server.key', certfile='server.crt')
    server.serve_forever()


def connect():
    """
    Test
    """
    # 2-tuple (host, port)
    address = ("127.0.0.1", 8443)

    # Socket
    current_socket = gevent.socket.socket()

    # Wrap
    current_socket = SSLSocket(current_socket)

    # Connect
    current_socket.connect(address)
    while True:
        print("Connected")
        gevent.sleep(1.0)


print("Serving")
gevent.spawn(serve)
print("Serving : done")

print("Connecting")
gevent.spawn(connect)

while True:
    print("Serving...")
    gevent.sleep(5.0)
```
********************
*** 2.7.9 code
********************
```
from __future__ import print_function
from gevent import pywsgi
import gevent

from gevent import monkey
from gevent._sslgte279 import SSLSocket

monkey.patch_all()


def on_connect(env, start_response):
    """
    Doc
    :param env:
    :param start_response:
    :return:
    """
    print("Connected")


def serve():
    """
    Test
    :return:
    """
    server = pywsgi.WSGIServer(('0.0.0.0', 8443), on_connect,
                               keyfile='server.key', certfile='server.crt')
    server.serve_forever()


def connect():
    """
    Test
    """

    # 2-tuple (host, port)
    address = ("127.0.0.1", 8443)

    # Socket
    current_socket = gevent.socket.socket()

    # Wrap
    current_socket = SSLSocket(current_socket)

    # Connect
    current_socket.connect(address)
    while True:
        print("Connected")
        gevent.sleep(1.0)

print("Serving")
gevent.spawn(serve)
print("Serving : done")

print("Connecting")
gevent.spawn(connect)

while True:
    print("Serving...")
    gevent.sleep(5.0)

```

**Comment by SuiperLolo** [comment]
After deeper look, i got a beautifull SSLError: [SSL] EVP lib (_ssl.c:581) at server side...
should be related to 
https://www.python.org/dev/peps/pep-0476/

Didn't find a way to disable it for tcp yet...

```
/home/test/.virtualenvs/env.g/bin/python2.7 /home/test/_devs/lab/gev/s.py
Serving
Serving : done
Connecting
Serving...
Traceback (most recent call last):
  File "/home/test/.virtualenvs/env.g/local/lib/python2.7/site-packages/gevent/greenlet.py", line 327, in run
    result = self._run(*self.args, **self.kwargs)
  File "/home/test/_devs/lab/gev/s.py", line 64, in connect
    current_socket.connect(address)
  File "/home/test/.virtualenvs/env.g/local/lib/python2.7/site-packages/gevent/_sslgte279.py", line 553, in connect
    self._real_connect(addr, False)
  File "/home/test/.virtualenvs/env.g/local/lib/python2.7/site-packages/gevent/_sslgte279.py", line 544, in _real_connect
    self.do_handshake()
  File "/home/test/.virtualenvs/env.g/local/lib/python2.7/site-packages/gevent/_sslgte279.py", line 511, in do_handshake
    return self._sslobj.do_handshake()
SSLError: ('_ssl.c:565: The handshake operation timed out',)
<Greenlet at 0x7fc6959ed230: connect> failed with SSLError

Traceback (most recent call last):
  File "/home/test/.virtualenvs/env.g/local/lib/python2.7/site-packages/gevent/greenlet.py", line 327, in run
    result = self._run(*self.args, **self.kwargs)
  File "/home/test/.virtualenvs/env.g/local/lib/python2.7/site-packages/gevent/server.py", line 102, in wrap_socket_and_handle
    ssl_socket = self.wrap_socket(client_socket, **self.ssl_args)
  File "/home/test/.virtualenvs/env.g/local/lib/python2.7/site-packages/gevent/_sslgte279.py", line 636, in wrap_socket
    ciphers=ciphers)
  File "/home/test/.virtualenvs/env.g/local/lib/python2.7/site-packages/gevent/_sslgte279.py", line 248, in __init__
    raise x
SSLError: [SSL] EVP lib (_ssl.c:581)
<Greenlet at 0x7fc6959ed230: <bound method WSGIServer.wrap_socket_and_handle of <WSGIServer at 0x7fc695785c10 fileno=5 address=0.0.0.0:8443>>(<socket at 0x7fc69541c890 fileno=6 sock=127.0.0.1:, ('127.0.0.1', 40083))> failed with SSLError

Serving...
KeyboardInterrupt
Traceback (most recent call last):
  File "/home/test/_devs/lab/gev/s.py", line 78, in <module>
    gevent.sleep(5.0)
  File "/home/test/.virtualenvs/env.g/local/lib/python2.7/site-packages/gevent/hub.py", line 82, in sleep
    hub.wait(loop.timer(seconds, ref=ref))
  File "/home/test/.virtualenvs/env.g/local/lib/python2.7/site-packages/gevent/hub.py", line 348, in wait
    result = waiter.get()
  File "/home/test/.virtualenvs/env.g/local/lib/python2.7/site-packages/gevent/hub.py", line 575, in get
    return self.hub.switch()
  File "/home/test/.virtualenvs/env.g/local/lib/python2.7/site-packages/gevent/hub.py", line 338, in switch
    return greenlet.switch(self)
KeyboardInterrupt

Process finished with exit code 1
```

**Comment by SuiperLolo** [comment]
note : this is with a self-signed certificate.

i tried to regen it (openssl req -new -x509 -days 365 -nodes -out cert.pem -keyout cert.pem)
& add it to root certs (update-ca-certificates --fresh), failed
even forcing stuff, no success...

```
    server = pywsgi.WSGIServer(('0.0.0.0', 8443), on_connect,
                               keyfile='server.key', certfile='server.crt',
                               ca_certs="/etc/ssl/certs/ca-certificates.crt",
                               cert_reqs=CERT_REQUIRED,
                               ssl_version=PROTOCOL_SSLv23)
```

At the end, with ca_certs / cert_reqs disabled or enabled :
- old certif + PROTOCOL_SSLv23 : SSLError: [SSL] EVP lib (_ssl.c:581)
- old certif + PROTOCOL_TLSv1 : SSLEOFError: EOF occurred in violation of protocol (_ssl.c:581)
- new certif + PROTOCOL_SSLv23 : SSLEOFError: EOF occurred in violation of protocol (_ssl.c:581)
- new certif + PROTOCOL_TLSv1 : SSLEOFError: EOF occurred in violation of protocol (_ssl.c:581)

running debian 8, OpenSSL 1.0.1k 8 Jan 2015, Python 2.7.9, gevent 1.0.2

SSL hell... after https, tcp....

**Comment by SuiperLolo** [comment]
Hey,

ok so, i was able to workaround it using a self-signed certificate with CA, with something like that

```
openssl genrsa -out rootCA.key 2048
openssl req -x509 -new -nodes -key rootCA.key -days 1024 -out rootCA.pem
sudo cp rootCA.pem /usr/local/share/ca-certificates/rootCA.crt
sudo update-ca-certificates
openssl genrsa -out device.key 2048
openssl x509 -req -in device.csr -CA rootCA.pem -CAkey rootCA.key -CAcreateserial -out device.crt -days 500
```

However, i am still not able to disable CA root verification at tcp level using gevent.
The global hack (for http layer) referenced in https://www.python.org/dev/peps/pep-0476/ is not working (logic, its for http i assume)...

Regards,

**Comment by SuiperLolo** [comment]
Forget the previous, it was on python 2.7.6...
still not working on 2.7.9

will come back...

**Comment by SuiperLolo** [comment]
Hey again,

sorry for last post, server mismatch :(

Go back to python 2.7.9.

For gevent client/server, i still got at server side :
```
site-packages/gevent/_sslgte279.py", line 248, in __init__
    raise x
SSLEOFError: EOF occurred in violation of protocol (_ssl.c:581)
<Greenlet at 0x7f2d291332d0: <bound method WSGIServer.wrap_socket_and_handle of <WSGIServer at 0x7f2d28ececd0 fileno=5 address=127.0.0.1:8443>>(<socket at 0x7f2d28b67950 fileno=6 sock=127.0.0.1:, ('127.0.0.1', 20563))> failed with SSLEOFError
```

Keeping the server running and checking with openssl s_client, i saw the connection establishing (tlsv1, with selfsigned, logic)

```
openssl s_client -connect 127.0.0.1:8443 -tls1 -key server.key -cert server.crt      
CONNECTED(00000003)
depth=1 C = US, ST = Oregon, L = Portland, O = Overlords, OU = IT, CN = Data Center Overlords, emailAddress = none@none.com
verify error:num=19:self signed certificate in certificate chain
verify return:0
---
Certificate chain
 0 s:/C=US/ST=Oregon/L=Portland/O=Overlords/OU=IT/CN=127.0.0.1/emailAddress=none@none.com
   i:/C=US/ST=Oregon/L=Portland/O=Overlords/OU=IT/CN=Data Center Overlords/emailAddress=none@none.com
 1 s:/C=US/ST=Oregon/L=Portland/O=Overlords/OU=IT/CN=Data Center Overlords/emailAddress=none@none.com
   i:/C=US/ST=Oregon/L=Portland/O=Overlords/OU=IT/CN=Data Center Overlords/emailAddress=none@none.com
---
Server certificate
-----BEGIN CERTIFICATE-----
MIIDkjCCAnoCCQCfBPT5UrvxvTANBgkqhkiG9w0BAQsFADCBkDELMAkGA1UEBhMC
VVMxDzANBgNVBAgMBk9yZWdvbjERMA8GA1UEBwwIUG9ydGxhbmQxEjAQBgNVBAoM
CU92ZXJsb3JkczELMAkGA1UECwwCSVQxHjAcBgNVBAMMFURhdGEgQ2VudGVyIE92
ZXJsb3JkczEcMBoGCSqGSIb3DQEJARYNbm9uZUBub25lLmNvbTAeFw0xNTA2MjUw
NTQzMDlaFw0xNjExMDYwNTQzMDlaMIGEMQswCQYDVQQGEwJVUzEPMA0GA1UECAwG
T3JlZ29uMREwDwYDVQQHDAhQb3J0bGFuZDESMBAGA1UECgwJT3ZlcmxvcmRzMQsw
CQYDVQQLDAJJVDESMBAGA1UEAwwJMTI3LjAuMC4xMRwwGgYJKoZIhvcNAQkBFg1u
b25lQG5vbmUuY29tMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEA8Qaz
nnN4iDdYiL8aWwmd8M42buQ67Cd2yJAjrcxVT/ajVLLCaOlwYUsoHrZsMPdJtJ1f
15IyH0A70NZls1i9Chsa3riS885jZe/osiV0F55OYlFPJcY9SmLleiiBZNlB8lgg
wX5ZCmnvJmMSx1yMIa/kymg9AHD409hGoR2qDoSAtPMGjuKIVktY+QOYAMcjtBO2
eGziv3cyYKj4q2xrdy+91HnY5tk0Gw1TUZvSbNL04UwVQ27wvRXI5OVY77MhEGip
nRhI4pBt8Quja6/acGyuNvZnBXP1Ue1PFTkTWh2HeK5wDUIgAmc0WIlTKsIgLtEP
46dhuvVvtDV0q7TWOQIDAQABMA0GCSqGSIb3DQEBCwUAA4IBAQDKeyh7cs9JTXNh
orV+jKntsffaH8YJPaDmLwQNAHULXvKwXTrV58iJNCdwdFdfkiVm0lvE0qRkLBzK
ATyFO7vPK7XBpF8j/5w6bU9iGsDuy0FJcBYflHhJWNys2c30XrCeTEoc560zWGGu
vHoHFvPA2sJDNtfx4P4se/uCvplbSePwyr3buBdqEWBLKFZ/BfXe7jMeSgetlSyK
qUAc9dP6gubLwLHrtK6svih0x06rCfiA+Fz21CyIfjaRPZfq0UcpuDgMwJ/+OscU
ZyhUb6f9FyXR96XWGIyb3TbeVIaiSrUKigmsmDF7cjYjeMIJDH/M+8+GU4Y3qjvX
8LA5a+w2
-----END CERTIFICATE-----
subject=/C=US/ST=Oregon/L=Portland/O=Overlords/OU=IT/CN=127.0.0.1/emailAddress=none@none.com
issuer=/C=US/ST=Oregon/L=Portland/O=Overlords/OU=IT/CN=Data Center Overlords/emailAddress=none@none.com
---
No client certificate CA names sent
---
SSL handshake has read 3541 bytes and written 1537 bytes
---
New, TLSv1/SSLv3, Cipher is ECDHE-RSA-AES256-SHA
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
SSL-Session:
    Protocol  : TLSv1
    Cipher    : ECDHE-RSA-AES256-SHA
    Session-ID: 9D875E0353728EAAD02608C14F5AA77251F51D1442652031DDE761703D3C62BF
    Session-ID-ctx: 
    Master-Key: BA50E2548D350D860D890EFF7026C64393A4683FE446160055C6D8CBC8C9D708FC74DC62362C4543B006C6705639CCDD
    Key-Arg   : None
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    TLS session ticket lifetime hint: 7200 (seconds)
    TLS session ticket:
    0000 - d8 d8 c3 f8 8b 4f 67 f9-5e 08 a6 34 59 43 cf c8   .....Og.^..4YC..
    0010 - 37 96 84 48 c9 10 ed 37-e3 b3 40 4c 67 9d 12 6d   7..H...7..@Lg..m
    0020 - d5 0a 0c bf 18 2b 79 f1-37 df ad f2 b6 1f cb fe   .....+y.7.......
    0030 - 1c a3 87 f4 9f b7 11 df-c9 91 12 cb 92 40 5f c0   .............@_.
    0040 - 2c 85 1c 72 aa 0a 39 d8-22 29 d1 22 48 60 2f db   ,..r..9.")."H`/.
    0050 - bb 80 1f e7 43 dd 65 a4-3f 50 5c 73 2c 51 07 fa   ....C.e.?P\s,Q..
    0060 - d6 65 2d 7e c1 b9 20 8b-ea 58 56 2e 01 1c 85 b0   .e-~.. ..XV.....
    0070 - be 7b e1 29 64 4f b5 cd-39 eb 5d f7 19 d7 a7 38   .{.)dO..9.]....8
    0080 - eb 52 06 27 d6 6e b3 77-46 1b 3c 45 e1 47 87 38   .R.'.n.wF.<E.G.8
    0090 - 69 15 1c 28 59 c7 b2 d9-36 22 f4 c2 7b 2a a2 05   i..(Y...6"..{*..
    00a0 - 52 d0 32 4e 0a 78 0e b7-f4 f7 01 f0 a2 e6 58 94   R.2N.x........X.
    00b0 - 31 40 f1 33 cb a4 93 4a-50 9c d1 2e d3 f9 ce e9   1@.3...JP.......
    00c0 - b1 be 41 68 5d 8a 8b 04-71 11 77 d6 6f 4d e6 a4   ..Ah]...q.w.oM..
    00d0 - 8c 42 8a f0 ab d7 eb 01-54 be 89 7d 13 56 56 17   .B......T..}.VV.
    00e0 - 7e 4b 26 3c 4d 56 3e e3-d1 f8 20 50 2f 18 7d 5b   ~K&<MV>... P/.}[
    00f0 - a1 c6 5b 93 41 6e 00 cc-25 78 fd 3c ff 8d 23 68   ..[.An..%x.<..#h
    0100 - a1 64 80 ca a0 de 95 ce-52 a3 cf 6f c0 57 dc b8   .d......R..o.W..
    0110 - f4 15 21 fb 23 b9 17 08-dd c5 fe 99 04 eb 93 14   ..!.#...........
    0120 - 49 dd 4a 9d c4 89 b5 b5-02 6f c0 6c 04 f3 64 ba   I.J......o.l..d.
    0130 - d9 49 4c 13 43 9b f3 6b-b2 a2 8c 73 7b e1 60 fb   .IL.C..k...s{.`.
    0140 - c5 f6 66 d3 8c 62 76 e7-46 88 5c c6 a4 ba eb f6   ..f..bv.F.\.....
    0150 - c9 f5 12 13 c2 14 64 ef-0c a7 06 ab bd 3e 76 9f   ......d......>v.
    0160 - 20 28 c9 01 1e da 8f 0c-c3 f7 cc 25 ad 29 c1 be    (.........%.)..
    0170 - 52 e1 68 63 9c 94 20 5e-1e 96 9d bd 18 2c a1 db   R.hc.. ^.....,..
    0180 - b2 0a 14 0c 2f 01 f7 95-bd 2b 11 c2 28 9f 3d 80   ..../....+..(.=.
    0190 - 8f 41 cc d0 07 74 23 47-a7 d1 b0 c0 e3 ca 9d 43   .A...t#G.......C
    01a0 - 1b fe 59 3e f1 02 df 61-52 57 75 12 0b 63 d2 fd   ..Y>...aRWu..c..
    01b0 - 49 c0 d8 3f 0c 14 c8 e6-db bc 74 23 bd e0 6b 79   I..?......t#..ky
    01c0 - 91 a1 3e df e6 be 3d 5e-45 76 b4 a0 b5 82 ed c4   ..>...=^Ev......
    01d0 - 77 e8 fc 24 a3 4e d8 05-10 e6 92 2e ee b5 79 fd   w..$.N........y.
    01e0 - e1 16 e9 84 d2 7f 4b 1e-02 4b c2 d2 83 a0 c3 8b   ......K..K......
    01f0 - a3 b0 36 04 73 94 78 83-11 aa 56 f9 28 8f 80 b3   ..6.s.x...V.(...
    0200 - f0 26 c4 c5 62 7b 90 33-1e ce 39 ae 14 b1 ec 09   .&..b{.3..9.....
    0210 - a8 b2 dd f9 67 c6 85 03-54 c5 c4 e1 6b b2 44 2e   ....g...T...k.D.
    0220 - 73 09 e0 d1 cf 39 d1 e5-b9 44 54 37 8a 1f 7e 68   s....9...DT7..~h
    0230 - ba 70 89 bb e3 bd 57 f3-93 29 34 90 a9 ff e7 1f   .p....W..)4.....
    0240 - bf e9 cc 86 87 3d 6d 4a-1f eb 43 00 db ff 1c 07   .....=mJ..C.....
    0250 - e7 4a 3e c2 89 b3 36 2d-76 50 cf cd e6 2f f9 ce   .J>...6-vP.../..
    0260 - 2c fb fd 04 33 d4 15 8a-9d a6 b7 45 57 ab ca e8   ,...3......EW...
    0270 - ce 38 83 62 22 81 0e 50-40 61 b2 b3 60 7d 12 36   .8.b"..P@a..`}.6
    0280 - 84 64 2d 4c 33 6e 31 12-35 9e e7 79 29 04 69 64   .d-L3n1.5..y).id
    0290 - 19 08 2e 37 b0 7b 8f a5-01 16 ce 9b af c7 cd d4   ...7.{..........
    02a0 - 94 0d 0d 4c 52 eb 34 62-7e aa 96 f2 40 9f 62 b3   ...LR.4b~...@.b.
    02b0 - ea 54 1f c8 31 fb a7 8c-63 20 fc 45 8b 2e 9f 6d   .T..1...c .E...m
    02c0 - 67 6d f0 7b e4 07 56 85-e3 36 72 42 34 23 76 8f   gm.{..V..6rB4#v.
    02d0 - c7 15 1c 1e 72 c9 41 e5-fc 0b fc 21 bb d0 61 19   ....r.A....!..a.
    02e0 - f9 12 3a 79 e5 2e d7 48-b8 3b 27 2f 26 36 0c 09   ..:y...H.;'/&6..
    02f0 - 9c de 25 1f ac ac ee ae-ef 85 f4 3e 05 6b af 02   ..%........>.k..
    0300 - d1 67 41 02 ca 79 38 15-01 aa 54 bd 77 ab 85 0c   .gA..y8...T.w...
    0310 - ad df 33 08 78 86 f7 0d-a1 95 b7 37 03 aa c3 5f   ..3.x......7..._
    0320 - 28 1e 80 90 9b 9d d4 f4-ac 43 a7 2f df df 26 75   (........C./..&u
    0330 - 4f e8 82 a5 db a3 e1 28-71 f6 2a 0b f1 53 ef 7b   O......(q.*..S.{
    0340 - c7 04 fa d6 37 07 b4 d2-1a ca b6 1b 06 23 8d 51   ....7........#.Q
    0350 - d9 9d 3f ee 36 c0 5a d1-e9 79 0f bf b6 d1 2c fa   ..?.6.Z..y....,.
    0360 - a0 23 4a 04 8b 87 b8 7f-5b 6a c1 9c e9 f6 d3 f4   .#J.....[j......
    0370 - 3b 8f c0 c1 63 19 30 16-f8 12 8d 57 24 8b 56 89   ;...c.0....W$.V.
    0380 - aa 3b b5 04 f0 82 c9 a1-ed e8 3a 1b e2 17 f2 45   .;........:....E
    0390 - a6 cc 99 4e 19 66 3e ef-53 87 6d 6c d5 af f4 ef   ...N.f>.S.ml....
    03a0 - 1c a2 91 3c ce 32 67 a5-56 36 b6 7d 64 be fc b9   ...<.2g.V6.}d...
    03b0 - 59 59 dd b7 44 4d 2d 3f-e2 3a d0 84 66 71 31 00   YY..DM-?.:..fq1.
    03c0 - f8 93 2c cf bc d1 63 a7-37 db 9d 63 40 c3 b3 3f   ..,...c.7..c@..?
    03d0 - d6 c5 ac 62 ef 7f 61 fa-99 4b b5 0b 96 2f 93 9d   ...b..a..K.../..
    03e0 - 4e 0f 2b 20 11 5d ee 3a-c8 21 1f 7b 24 03 d7 66   N.+ .].:.!.{$..f
    03f0 - e7 03 fc a5 51 37 28 87-cb 56 3e ae 4b a7 91 79   ....Q7(..V>.K..y
    0400 - ec c1 4d 0d 74 3f 3e 47-99 da ff d8 5c 60 04 88   ..M.t?>G....\`..
    0410 - 17 9e ad 47 32 21 8f e5-72 4f 24 59 a4 ae 20 10   ...G2!..rO$Y.. .
    0420 - 62 42 ff b8 8b e0 2d f3-68 57 78 6d 13 e6 89 4e   bB....-.hWxm...N
    0430 - 29 c2 82 fb 95 55 30 42-6b e4 34 95 6a aa 08 ac   )....U0Bk.4.j...

    Start Time: 1435218725
    Timeout   : 7200 (sec)
    Verify return code: 19 (self signed certificate in certificate chain)
---
^C
$ 
```

**Comment by Champax** [comment]
It seems i can repro that with gevent 1.1a1.

I am not sure if it is a gevent issue or python issue or a cert issue.
There are bunch of posts related to SSLEOFError: EOF occurred in violation of protocol :(

**Comment by Champax** [comment]
Hum, may be related to #594
will do more tests when i have time

**Comment by jamadden** [comment]
If you make `from gevent import monkey; monkey.patch_all()` the *first* line of your script, does it work for you under Python 2.7.9 (if you're going to monkey patch, doing so as the first line is always recommended)? If so, you may be hitting the same thing as #597.

**Comment by SuiperLolo** [comment]
Hello,

i found it yesterday, but i had no time to update the ticket : i confirm the issue with the imports order.

These imports raise the error "SSLError: [Errno 8] _ssl.c:510: EOF occurred in violation of protocol
<Greenlet at 0x7f9b4d249190: _close_when_done(<socket at 0x7f9b4d254490 fileno=5 sock=127.0.0.1:, ('127.0.0.1', 62050))> failed with SSLError"

```
from __future__ import print_function

from gevent.server import StreamServer

from _ssl import CERT_REQUIRED, PROTOCOL_SSLv23, PROTOCOL_TLSv1, CERT_OPTIONAL
from ssl import SSLSocket
import gevent

from gevent import monkey
monkey.patch_all()
```

These imports works fine
```
from __future__ import print_function

from gevent import monkey
monkey.patch_all()

from gevent.server import StreamServer

from _ssl import CERT_REQUIRED, PROTOCOL_SSLv23, PROTOCOL_TLSv1, CERT_OPTIONAL
from ssl import SSLSocket
import gevent
```

**Comment by jamadden** [comment]
Fix released in 1.1a2; an import-order workaround often works for 1.0.x.
